src/core/server/saved_objects/migrationsv2/migrations_state_action_machine.ts

```diff
@@ -183,9 +183,13 @@ export async function migrationStateActionMachine({
         );
       }
 
-      throw new Error(
+      const newError = new Error(
         `Unable to complete saved object migrations for the [${initialState.indexPrefix}] index. ${e}`
       );
+
+      // restore error stack to point to a source of the problem.
+      newError.stack = `[${e.stack}]`;
```

Comment:
nit: it makes sense to keep the stack of the original error, but maybe it's simpler to override `e.message` instead of constructing a new error?

Reply:
IMO it's a bit more clear when you search by an error message.